### ISSUE TITLE
Noob install question

### ISSUE THREAD

**Issue by NNois** [Aeoll/ae_SVG#1]
Hi, sorry but I get that when I add thoses lines when I open the python shell
(step 3 of your install instructions)
`>>> import sys
>>> sys.append("C:\Python27\Lib\site-packages")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: 'module' object has no attribute 'append'`

using 123 file don't work on windows because there is no 123.py but a 123.cmd...

So i've just copied the svg folder in the one of the houdini install dir and it works.
Is that correct ?

**Comment by Aeoll** [comment]
Hi,

Yes that's correct - i need to tweak the instructions ! Sorry for the
confusion

Nick

**Comment by Benra1** [comment]
Hi Noob Install Question as well.   I am having the worst time following the install instructions Nick and im hoping you can set me straight.  
Ive been at it for 4 straight hours and still cant get a solve. 
I'm not Python literate so ive been bumbling ahead. 
I've sucessfully got to the point of installing pip.  
But cannot get pip to install svg.path to my houdini/libs folder.  
do i exactly copy your instructions into the shell (windows) ? eg
pip install svg.path --target [C:/Program Files/Side Effects Software/Houdini 16.5.323/python27/libs]

i get the error return
Invalid requirement: 'Files/Side'
Traceback (most recent call last):
  File "c:\program files\side effects software\houdini 16.5.323\python27\lib\site-packages\pip\req\req_install.py", line 82, in __init__
    req = Requirement(req)
  File "c:\program files\side effects software\houdini 16.5.323\python27\lib\site-packages\pip\_vendor\packaging\requirements.py", line 96, in __init__
    requirement_string[e.loc:e.loc + 8]))
InvalidRequirement: Invalid requirement, parse error at "'/Side'"

tried without the --target and without [ ] but still no happy face.
going a bit nuts here. 
have to give up on 4 hours of effort and its past midnight now. gar.

**Comment by Aeoll** [comment]
Hi Benra,

Sorry for the struggles.
The first thing to do would be to try running 'pip install svg.path' . If you still see an error, there is a problem with your python27 or pip install and you should try reinstalling these from scratch.

If there is no error svg.path has installed correctly. You can copy it from its default install location into the houdini16.0/python2.7libs folder. For me this default install location is C:\Python27\Lib\site-packages\svg. Copy the 'svg' folder and don't forget to create the __init__.py file inside.

**Comment by Benra1** [comment]
for the sake of sanity and the remote possibility that adding more will help and not hinder .... 
this is where ive got to .

I get mixed messages back... but i figure its failed with all that red text. 


PS C:\users\ben> pip install svg.path --target [C:/Program_Files/Side_Effects_Software/Houdini_16.5.323/python27/libs]
Collecting svg.path
  Using cached svg.path-2.2.tar.gz
Collecting setuptools (from svg.path)
  Using cached setuptools-38.5.1-py2.py3-none-any.whl
Building wheels for collected packages: svg.path
  Running setup.py bdist_wheel for svg.path ... error
  Complete output from command "c:\program files\side effects software\houdini 16.5.323\python27\python.exe" -u -c "import setuptools, tokenize;__file__='c:\\users\\ben\\appdata\\local\\temp\\pip-build-dyujux\\svg.path\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d c:\users\ben\appdata\local\temp\tmpva3tdmpip-wheel- --python-tag cp27:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help

  error: invalid command 'bdist_wheel'

  ----------------------------------------
  Failed building wheel for svg.path
  Running setup.py clean for svg.path
Failed to build svg.path
Installing collected packages: setuptools, svg.path
  Running setup.py install for svg.path ... done
Successfully installed setuptools-38.5.1 svg.path-2.2
Exception:
Traceback (most recent call last):
  File "c:\program files\side effects software\houdini 16.5.323\python27\lib\site-packages\pip\basecommand.py", line 215, in main
    status = self.run(options, args)
  File "c:\program files\side effects software\houdini 16.5.323\python27\lib\site-packages\pip\commands\install.py", line 388, in run
    ensure_dir(options.target_dir)
  File "c:\program files\side effects software\houdini 16.5.323\python27\lib\site-packages\pip\utils\__init__.py", line 83, in ensure_dir
    os.makedirs(path)
  File "c:\program files\side effects software\houdini 16.5.323\python27\lib\os.py", line 150, in makedirs
    makedirs(head, mode)
  File "c:\program files\side effects software\houdini 16.5.323\python27\lib\os.py", line 150, in makedirs
    makedirs(head, mode)
  File "c:\program files\side effects software\houdini 16.5.323\python27\lib\os.py", line 150, in makedirs
    makedirs(head, mode)
  File "c:\program files\side effects software\houdini 16.5.323\python27\lib\os.py", line 150, in makedirs
    makedirs(head, mode)
  File "c:\program files\side effects software\houdini 16.5.323\python27\lib\os.py", line 150, in makedirs
    makedirs(head, mode)
  File "c:\program files\side effects software\houdini 16.5.323\python27\lib\os.py", line 157, in makedirs
    mkdir(name, mode)
WindowsError: [Error 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\users\\ben\\[C:'

**Comment by Benra1** [comment]
oh .. hey .. thanks for the quick reply ..  I will delete that error message comment.. its just in the way of things here. 
I will try what you suggest

**Comment by Benra1** [comment]
sweet,.. yes.. its there.  thanks 
But i dont have a python2.7libs folder showing . 
i have a python27/lib folder and a python27/libs folder
/libs has a few *.lib files and /lib has many folders with many files .
i will try placing into /libs .

**Comment by Benra1** [comment]
Thanks Nick !  That did the trick...  its half past midnight but at least i got it working. (5hours of my life just rolled by).

**Comment by Aeoll** [comment]
Glad you got it working. Python can be very annoying...! Hopefully the HDA is useful to you

**Comment by Benra1** [comment]
I'm running over some of your other amazing tools in the AEMaster set.  fantastic work Nick.  Some of them are intriguing and interesting, yet im not sure how to implement half of them.  (sorry, off this specific tool topic posting here).   is there any documentation or anything available to understand a particular tool in context?  cheers.

**Comment by Aeoll** [comment]
The documentation is admittedly a bit sparse. I have some videos on vimeo eg https://vimeo.com/204258573 for the older tools, which are coupled with HIPs in the Aelib examples/ directory.

I also have a Patreon page where I post about individual tools in more detail with commented scene files (and also share extra tools which aren't available for free!) https://www.patreon.com/aeoll

**Comment by Benra1** [comment]
much appreciated Nick..  will follow up with you at patreon.  cheers.

**Comment by JohnGemstone** [comment]
Hey Nick, just wanted to add how I got this installed as I couldn't get the methods here to work for me. Instead of installing the module from my own instance of python, I installed directly from houdini's instance instead. Running Windows 10.

1. download [pip](https://bootstrap.pypa.io/get-pip.py) and save to `C:\Program Files\Side Effects Software\Houdini 17.0.352\python27`

2. navigate to that folder with cmd.exe and enter `python2.7.exe get-pip.py --user`

3. once pip is installed enter `python2.7.exe -m pip install svg.path --user`

Thanks!
